### PR TITLE
Apply will fail with managed fields + tests

### PR DESCRIPTION
**Pull request by joshmsamuels**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR will make apply fail when managed fields is specified in an objects Metadata. It was a requested modification as a part of server-side apply.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by joshmsamuels** [comment]
@apelisse Are release notes needed for this change?

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go [hunk 1]
```diff
@@ -149,9 +149,11 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 // Apply is used when server-side apply is called, as it merges the
 // object and update the managed fields.
 func (f *FieldManager) Apply(liveObj runtime.Object, patch []byte, fieldManager string, force bool) (runtime.Object, error) {
-	// If the object doesn't have metadata, apply isn't allowed.
-	if _, err := meta.Accessor(liveObj); err != nil {
+	// If the object doesn't have metadata or managed fields is not empty, apply isn't allowed.
+	if objMeta, err := meta.Accessor(liveObj); err != nil {
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go]: `liveObj` is the object as it is on the server, the server is allowed to have managed fields!

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go [hunk 2]
```diff
@@ -439,3 +440,75 @@ func BenchmarkRepeatedUpdate(b *testing.B) {
...
+					Manager: "test",
+				},
+			},
+		},
+	}
+
+	_, err := f.Apply(obj, []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Pod",
+		"metadata": {
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go]: What you should be testing is that this object (the one in YAML, which is the one that is applied) has or has not the `managedFields`.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go (hunk 1 above)
Comment (joshmsamuels) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go]: So how can I check for managed fields? Cant the patch take many forms?

Comment (joshmsamuels) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go]: Ignore my last comment, I found my mistake

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go [hunk 3]
```diff
@@ -87,18 +87,6 @@ func TestApplyStripsFields(t *testing.T) {
 			"uid": "b",
 			"clusterName": "b",
 			"generation": 0,
-			"managedFields": [{
```
[the remark this answers is not in the document]
Comment (joshmsamuels) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go]: Thats a good idea. I didnt know it was called in update. I updated the test

**Comment by apelisse** [comment]
That test looks good,
/lgtm

**Comment by liggitt** [comment]
isn't this required for beta / 1.16?

**Comment by apelisse** [comment]
That'd be great actually

**Comment by liggitt** [comment]
Since this is adjusting API behavior, I think it needs to be included in 1.16. Let's go ahead and get a cherry-pick against the release-1.16 branch opened.
/milestone v1.16

cc @lavalamp for visibility

**Comment by justaugustus** [comment]
/priority critical-urgent
cc: @lachie83 @kubernetes/release-team

**Comment by lavalamp** [comment]
Uh, yeah, this is very important. Thanks!

/lgtm
/approve

**Pull request merged**